xray/client_test.go

```diff
@@ -225,3 +228,42 @@ func TestRoundTrip_reuse_datarace(t *testing.T) {
 	}
 	wg.Wait()
 }
+
+func TestRoundTrip_http2_datarace(t *testing.T) {
```

Comment:
Nitpick.... Could we rename this function to be "TestRoundTripHttp2Datarace"? Since we haven't used underscore in naming our functions, it would be good to keep consistency.